src/ElasticRepository.js

```diff
@@ -87,8 +88,10 @@ class ElasticRepository {
   }
   async search(body) {
     try {
+      const index=body.index
+      delete body.index;
       let search = await this.client.search({
-        index: this.baseIndex,
+        index: index|| this.baseIndex,
```

Comment:
Why not use the constructor itself to set the baseIndex?

Reply:
we want to keep the ability to query index dynamic.